### DISCUSSION TITLE
BYO Dockerfile ?

### DISCUSSION THREAD

**Discussion by gotson** [jreleaser/jreleaser#1138] [Q&A]
Hello,

when reading https://jreleaser.org/guide/latest/reference/packagers/docker.html my understanding is that JReleaser will create the `Dockerfile`, then use it to create the Docker image.

I already have a `Dockerfile`, and i don't want JReleaser to build it for me, is it possible to use my existing `Dockerfile` with JReleaser?

Thanks

**Comment by aalmiray** [comment]
Yes, you can set an existing Dockerfile as a template (even if it does not require any variable placeholders). See the `templateDirectory` property of the docker packager. 

JReleaser does this for its slim and alpine images:

https://jreleaser.org/guide/latest/reference/packagers/docker.html#_templates

https://github.com/jreleaser/jreleaser/tree/main/src/jreleaser/distributions/jreleaser-standalone/docker

**Reply by gotson** [reply]
> JReleaser does this for its slim and alpine images

Those are still `.tpl` files though. Let's say I have an existing `Dockerfile` somewhere in my project, could JReleaser make use of it?

**Reply by aalmiray** [reply]
Yes, you can. For the time being the file must be named Dockerfile.tpl, even if it does not use any variable placeholders. This ensures your file is used instead of the default template.

**Reply by gotson** [reply]
I don't quite understand the `commit` part of the Docker packager. What does this do ? And can this be disabled?

**Reply by gotson** [reply]
I've set `packagers.docker.repository.active=NEVER`, and i specified a Docker template, but i am getting the following error:

```
[ERROR] project.website must not be blank
[ERROR] project.docsUrl must not be blank
[ERROR] project.license must not be blank
[ERROR] project.authors must not be blank
```

Since i'm not using the default template, those should not be required. WDYT?

**Reply by gotson** [reply]
I also encounter the following:

```
[ERROR] registry.DEFAULT.password must not be blank. Configure a value using the Gradle DSL, or define a JRELEASER_DOCKER_DEFAULT_PASSWORD environment variable, or define a key/value pair in /Users/groebroeck/.jreleaser/config.properties with key = JRELEASER_DOCKER_DEFAULT_PASSWORD
```

Is that because JReleaser uses docker-java, and thus cannot benefit from existing login, for example from https://github.com/docker/login-action ?

**Reply by aalmiray** [reply]
> I've set `packagers.docker.repository.active=NEVER`, and i specified a Docker template, but i am getting the following error:
> 
> ```
> [ERROR] project.website must not be blank
> [ERROR] project.docsUrl must not be blank
> [ERROR] project.license must not be blank
> [ERROR] project.authors must not be blank
> ```
> 
> Since i'm not using the default template, those should not be required. WDYT?

There's no way (ahead of time) to tell if these properties may or may not be used by templates. JReleaser errs on the side of caution and expects them to be defined.

**Reply by gotson** [reply]
@aalmiray i see that `docsUrl` and `website` are marked as deprecated, is that normal ?

**Reply by aalmiray** [reply]
Yes, the preferred way to define this values is now via `homepage` and `documentation` inside `project.links`.
